Perusahaan X ingin meningkatkan penjualan mereka. Dari data penjualan sebelumnya menunjukkan bahwa penjualan rata-rata yaitu $100 per transaksi. Setelah melakukan training kepada pekerja sales, data penjualan terbaru/'TransactionAmount' (yang diambil dari 25 sampel pekerja sales) 

In [4]:
import pandas as pd
import numpy as np

In [5]:
#Import Data penjualan setelah Melakukan Training Sales
transaction_data = [100, 150, 50, 100, 130, 120, 100, 85, 70, 150, 150, 120, 50, 100, 100, 140, 90, 150, 50, 90, 120, 100, 110, 75, 65]

In [6]:
#Membuat DataFrame menggunakan Pandas
df = pd.DataFrame(transaction_data, columns=['TransactionAmount'])

In [7]:
print(df)


    TransactionAmount
0                 100
1                 150
2                  50
3                 100
4                 130
5                 120
6                 100
7                  85
8                  70
9                 150
10                150
11                120
12                 50
13                100
14                100
15                140
16                 90
17                150
18                 50
19                 90
20                120
21                100
22                110
23                 75
24                 65


In [8]:
#Mencari Mean, Median & Mode
mean_value = df['TransactionAmount'].mean()
median_value = df['TransactionAmount'].median()
mode_value = df['TransactionAmount'].mode()[0]

print(f"Mean: ${mean_value}")
print(f"Median: ${median_value}")
print(f"Mode: ${mode_value}")

Mean: $102.6
Median: $100.0
Mode: $100


In [9]:
#menghitung beberapa statistik deskriptif (range, variance, dan standard deviation)
range_value = df['TransactionAmount'].max() - df['TransactionAmount'].min()
variance_value = df['TransactionAmount'].var()
std_dev_value = df['TransactionAmount'].std()

print(f"Range: ${range_value}")
print(f"Variance: ${variance_value}")
print(f"Standard Deviation: ${std_dev_value}")

Range: $100
Variance: $1012.7499999999997
Standard Deviation: $31.82373328193912


In [10]:
from scipy import stats


In [11]:
#Data Penjualan Sebelum Melakukan Training Pada Sales
population_mean = 100


In [12]:
#Mencari t statistics & P-Value dengan mengeksekusi one sample t-test
t_statistic, p_value = stats.ttest_1samp(df['TransactionAmount'], population_mean)

print(f"T-Statistic: {t_statistic}")
print(f"P-Value: {p_value}")

T-Statistic: 0.4085001556802841
P-Value: 0.6865284813438117


P-Value 0.68 > dari Alpha (0.05) Maka Gagal untuk menolak H0
Maka Konklusi dari hasil ini adalah Training Sales merupakan Kegiatan yang kurang efektif dalam meningkatkan penjualan

In [14]:
#Untuk tes konklusi hasil t statistic kita bandingkan dengan critical region
#H1 lebih besar maka grafik akan condong ke kanan
stats.t.ppf(1-0.05, 23)

1.7138715277470473

Critical region berada pada 1.71 sedangkan t statistik ada pada 0.40, Maka t statistik tidak ada di critical region(Gagal menolak H0)